### DISCUSSION TITLE
PeerGone state

### DISCUSSION THREAD

**Discussion by Antiarchitect** [ntex-rs/ntex-mqtt#102] [Q&A]
Can you please help me to understand is PeerGone is an artificial "control packet" that comes only after Disconnect packet is sent or it can appear by other reasons?

**Comment by fafhrd91** [comment]
PeerGone is sent after socket disconnects

**Reply by Antiarchitect** [reply]
Can it appear before Disconnect is sent? Like abnormal termination?

**Reply by fafhrd91** [reply]
no, it cannot. you should always get peergone message. disconnect on other hand is reaction for Disconnect frame from peer

**Reply by Antiarchitect** [reply]
So if the peer socker was closed on TCP level there will no be Disconnect, but PeerGone will still appear, or I'm misunderstanding something?

**Reply by fafhrd91** [reply]
that is correct, you’ll get peergone

**Reply by Antiarchitect** [reply]
Thank you so much @fafhrd91
